### DISCUSSION TITLE
vercel.json SPA rewrite not working locally with "vercel dev" command

### DISCUSSION THREAD

**Discussion by thomasmattheussen** [vercel/community#4548] [Help]
### Summary

I made an SPA with vite and solidJS. I've added a vercel.json file with following contents as per the vercel documentation.

```
{
    "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}
```
It works fine in production. But I can't get it to work locally when running the `vercel dev` command. In my console, I get these errors

```
Loading module from “http://localhost:3000/@vite/client” was blocked because of a disallowed MIME type (“”).
Loading failed for the module with source “http://localhost:3000/@vite/client”.
Loading module from “http://localhost:3000/src/index.jsx” was blocked because of a disallowed MIME type (“”).
Loading failed for the module with source “http://localhost:3000/src/index.jsx”. 
```

### Steps to Reproduce

vercel.json

```
{
    "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}
```

vite.config.js
```
import { defineConfig } from 'vite';
import solidPlugin from 'vite-plugin-solid';

export default defineConfig({
  plugins: [
    solidPlugin(),
  ],
  server: {
    port: 3000,
  },
  build: {
    target: 'esnext',
  },
});
```

**Comment by jakst** [comment]
Facing the same issue. I need the rewrite when running on Vercel, but it breaks local dev through `vercel dev` because it redirects all dev server requests to the index.html file.

**Reply by thomasmattheussen** [reply]
I've fixed it by creating a `vercel.local.json` file with following contents:

```json
{
    "comment": "This is a bogus file because of https://github.com/orgs/vercel/discussions/4548"
}
```

and then running `vercel dev` with the `-A` option:

```
vercel -A vercel.local.json dev
```

Does the job.

**Comment by thomasmattheussen** [comment]
related: https://github.com/vercel/vercel/discussions/4538

**Reply by jakst** [reply]
Hah, looks like I was in that thread reproducing the issue way back in 2020 as well 😅

**Comment by thomasmattheussen** [comment]
also related: https://github.com/vercel/vercel/discussions/7100
